### PR TITLE
Use promises and ES2015ify tests

### PR DESCRIPTION
**Pull request by kevva**
Also adds XO linting and a small Travis tweak.

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -25,31 +24,10 @@ if (cli.input.length === 0) {
...
-		process.exit(1);
-	}
-
-	trash(files, function (err) {
-		if (err) {
-			console.error(err.message);
-			process.exit(1);
-		}
+Promise.all(cli.input.map(function (el) {
+	return pify(pathExists, Promise)(el).then(function (exists) {
```
Comment (arthurvr) [inline cli.js]: `path-exists` has already been updated to use a Promise API so I don't think there's a reason to `pify` it?

Comment (kevva) [inline cli.js]: Ah, sweet. I'll bump it.

File: package.json [hunk 2]
```diff
@@ -39,16 +39,23 @@
...
-    "win-trash": "^1.0.1",
-    "xdg-trash": "^1.0.0"
+    "win-trash": "^2.0.0",
+    "xdg-trash": "^2.0.1"
   },
   "devDependencies": {
-    "mocha": "*",
-    "path-exists": "^1.0.0"
+    "ava": "*",
+    "path-exists": "^1.0.0",
```
Comment (SamVerschueren) [inline package.json]: Is it necessary to add `path-exists` as dev dependency as well? It's already a 'normal' dependency.

File: test/test.js [hunk 3]
```diff
@@ -1,75 +1,75 @@
...
-		assert(!pathExists.sync('123'));
-		cb();
-	});
+	]);
+
+	t.false(pathExists.sync('fixture'));
+	t.false(pathExists.sync('fixture2'));
+	t.false(pathExists.sync(weirdName));
+	t.false(pathExists.sync('123'));
+	t.end();
```
Comment (SamVerschueren) [inline test/test.js]: `t.end()` is not needed in an `async`/`await` test.

File: test/test.js [hunk 4]
```diff
@@ -1,75 +1,75 @@
...
-		assert.ifError(err);
-		assert(!pathExists.sync('fdir'));
-		assert(!pathExists.sync(321));
-		cb();
-	});
+	]);
+
+	t.false(pathExists.sync('fdir'));
+	t.false(pathExists.sync(321));
+	t.end();
```
Comment (SamVerschueren) [inline test/test.js]: Same as above

File: test/test.js [hunk 5]
```diff
@@ -1,75 +1,75 @@
...
+	await pify(childProcess.execFile, Promise)(cli, [
 		'foobar',
 		'unicorn'
-	], function (err) {
-		assert.ifError(err);
-		cb();
-	});
+	]);
+
+	t.end();
```
Comment (SamVerschueren) [inline test/test.js]: Same as above

File: package.json (hunk 2 above)
Comment (kevva) [inline package.json]: Yeah, it gives you a good overview of what's needed for the tests to run. If you suddenly doesn't need `path-exists` anymore in your regular dependencies and you forget that your tests depends on it they will fail.

Comment (SamVerschueren) [inline package.json]: Thanks for the clarification :).

**Comment by SamVerschueren** [comment]
Looks good

**Pull request closed without merging**